materialize-rockset/driver.go

```diff
@@ -35,40 +37,100 @@ func (c *config) Validate() error {
 			return fmt.Errorf("missing '%s'", req[0])
 		}
 	}
+	return nil
+}
+
+// eventTimeInfo is copied from rtypes.EventTimeInfo and modified to customize the JSON schema.
+type eventTimeInfo struct {
+	Field    string  `json:"field" jsonschema:"title=Field Name,description=Name of the field containing the event time"`
+	Format   *string `json:"format,omitempty" jsonschema:"title=Format,description=Format of the time field,enum=milliseconds_since_epoch,enum=seconds_since_epoch"`
+	TimeZone *string `json:"time_zone,omitempty" jsonschema:"title=Timezone,description=Default timezone, in IANA format"`
+}
+
+func (e *eventTimeInfo) Validate() error {
+	if e.Field == "" {
+		return fmt.Errorf("Event Time Info: Field Name is empty")
+	}
+	if e.TimeZone != nil {
+		if _, err := time.LoadLocation(*e.TimeZone); err != nil {
+			return fmt.Errorf("Event Time Info: invalid Timezone: %w", err)
+		}
+	}
+
+	var validFormats = map[string]bool{
+		"milliseconds_since_epoch": true,
+		"seconds_since_epoch":      true,
+	}
+	if e.Format != nil {
+		if !validFormats[*e.Format] {
+			return fmt.Errorf("Event Time Info: Format is invalid")
+		}
+	}
+	return nil
+}
+
+// fieldPartition was copied from rtypes.FieldPartition and modified both to customize the json
+// schema and to remove unnecessary fields. Turns out that all the fields except for `FieldName` are
+// seemingly unnecessary, beucause the only supported `type` is `AUTO`, and the
+// [docs](https://rockset.com/docs/rest-api/#createcollection) say that the `keys` are not needed if
+// type is `AUTO`.
+type fieldPartition struct {
+	FieldName *string `json:"field_name,omitempty" jsonschema:"title=Field Name,description=The name of a field\u002C parsed as a SQL qualified name"`
```

Comment:
Yes. Go doesn't have a way to escape commas in struct tag values, so we're relying on JSON string escapes. Although, it turns out that the UI doesn't even render this description, anyway because of the way it's nested in an array :facepalm: